src/push-build-information.ts

```diff
@@ -50,63 +34,43 @@ export async function pushBuildInformation(
   const repoUri = `https://github.com/${context.repo.owner}/${context.repo.repo}`
 
   const build = {
-    BuildEnvironment: 'GitHub Actions',
-    BuildNumber: runId.toString(),
-    BuildUrl: `${repoUri}/actions/runs/${runId}`,
-    Branch: branch,
-    VcsType: 'Git',
-    VcsRoot: `${repoUri}`,
-    VcsCommitNumber: context.sha
+    PackageId: '',
+    Version: parameters.version,
+    OctopusBuildInformation: {
+      BuildEnvironment: 'GitHub Actions',
+      BuildNumber: runId.toString(),
+      BuildUrl: `${repoUri}/actions/runs/${runId}`,
+      Branch: branch,
+      VcsType: 'Git',
+      VcsRoot: `${repoUri}`,
+      VcsCommitNumber: context.sha
+    }
   }
 
-  info(`Writing build information to buildInformation.json`)
-  const jsonContent = JSON.stringify(build)
-  await fsPromises.writeFile('buildInformation.json', jsonContent)
-
   if (parameters.debug) {
-    info(`Build Information: ${jsonContent}`)
+    info(`Build Information:\n${JSON.stringify(build, null, 2)}`)
   }
 
-  const args = getArgs(parameters)
-  args.push('--file=buildInformation.json')
-
-  const options: ExecOptions = {
-    listeners: {
-      stdline: (line: string) => {
-        if (line.length <= 0) return
-
-        if (line.includes('Octopus Deploy Command Line Tool')) {
-          const version = line.split('version ')[1]
-          info(`🐙 Using Octopus Deploy CLI ${version}...`)
-          return
-        }
-
-        if (line.includes('Handshaking with Octopus Server')) {
-          info(`🤝 Handshaking with Octopus Deploy`)
-          return
-        }
-
-        if (line.includes('Authenticated as:')) {
-          info(`✅ Authenticated`)
-          return
-        }
+  try {
+    const client = await getOctopusClient(parameters)
+    const buildInfoUriTmpl = client.getLink('BuildInformation')
 
-        if (line === 'Push successful') {
-          info(`🎉 Push successful!`)
-          return
-        }
+    for (const packageId of parameters.packages) {
+      info(
+        `Pushing build information for package '${packageId}' version '${parameters.version}'`
+      )
 
-        info(line)
-      }
-    },
-    silent: true
-  }
+      build.PackageId = packageId
+      await client.post(buildInfoUriTmpl, build, {
```

Comment:
I wanted to use our `BuildInformationRepository` here, but unfortunately it is out-of-date with what the current Octopus API is expecting to be passed to it. So, instead of delaying getting this action built I just use a straight `post` here until we can get to updating [OctopusDeploy/api-client.ts](https://github.com/OctopusDeploy/api-client.ts) with the new shape of resource it is expecting.